rust-runtime/aws-smithy-runtime/src/client/retries/strategy/standard.rs

```diff
@@ -147,7 +147,15 @@ impl StandardRetryStrategy {
                         // The initial attempt shouldn't count towards backoff calculations, so we subtract it
                         request_attempts - 1,
                     );
-                    Ok(Duration::from_secs_f64(backoff).min(retry_cfg.max_backoff()))
+                    match Duration::try_from_secs_f64(backoff) {
+                        Ok(duration) => Ok(duration.min(retry_cfg.max_backoff())),
+                        Err(e) => {
+                            tracing::warn!(
+                                "could not create `Duration` for exponential backoff: {e}"
+                            );
+                            Err(ShouldAttempt::No)
```

Comment:
This seems logical to me but just to be sure we've done our due diligence I'm going to poke at this.

1. Why does it make sense to not attempt instead of just capping the backoff to `retry_cfg.max_backoff()`?
2. I glanced through the SEP I don't see any guidance on this scenario, have we asked other SDKs what they do to ensure there is consistency? It may be worth updating the SEP if this is a grey area.


----

Separate question, if we are setting 100 attempts what does the token bucket look like here? I kind of would have thought we'd be at least getting close to exceeding our quotas.

Reply:
Yeah, I think this would be better:

`Duration::try_from_secs_f64(backoff).unwrap_or(Duration::MAX).min(retry_cfg.max_backoff())`

Reply:
After internal discussion, we learned it's still debatable whether we should continue making attempts. So we're not going to deliberately make that decision in this PR (and not making that decision doesn't make anythings worse than it is today). We can still make an improvement in that area once we've reached a consensus.

This PR allows customers to perform retries even when overflow occurs, which is updated in 54ad1f0.

Reply:
> Duration::try_from_secs_f64(backoff).unwrap_or(Duration::MAX).min(retry_cfg.max_backoff())

This one-liner is tempting, but there was some clarification in the design that jitter also needs to be applied to `max_backoff`. To make everything contained in one function (worrying about overflow, whether it should fallback to `max_backoff`, and for the ease of testing), I put things the way I did in the above commit. Let me know if you can think of a different way.